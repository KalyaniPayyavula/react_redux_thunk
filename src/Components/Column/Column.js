import PropTypes  from "prop-types"
import { styled } from "@mui/system"

export const Column = ({children, dataTestId}) =>{
    const ColumnStyle = styled('div')({
        display : "flex",
        flexDirection : "column",
        marginLeft : "35%",
        width  : "20%"
    })
    return(
        <ColumnStyle data-testid = {dataTestId}>{children}</ColumnStyle>
    )
}

Column.propTypes = {
    dataTestId: PropTypes.string.isRequired,
    children : PropTypes.array.isRequired  
}