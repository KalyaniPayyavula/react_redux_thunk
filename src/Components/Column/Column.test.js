/* eslint-disable no-undef */
/* eslint-disable react/no-children-prop */
import { render, screen } from "@testing-library/react";
import { Column } from "./Column";

test("Column component", () => {
	render(
		<Column dataTestId="test-column">
			<p>{["one", "two"]}</p>
		</Column>
	);

	expect(screen.getByTestId("test-column").textContent).toBe("onetwo");
});

test("Snapshot for Column component", () => {
	render(
		<Column
			children={<p>childOne</p>}
			label="test"
			dataTestId="test-column"
		/>
	);

	expect(screen.getByTestId("test-column").textContent).toMatchSnapshot();
});
