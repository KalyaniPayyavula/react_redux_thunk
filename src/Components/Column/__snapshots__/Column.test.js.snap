// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for Column component 1`] = `"childOne"`;
